search/includes/classes/class-search.php

```diff
@@ -1062,12 +1062,19 @@ public function ep_handle_failed_request( $request, $response, $query, $statsd_p
 
 	/**
 	 * Given an ElasticPress query object, strip out anything that shouldn't be logged
+	 * Or format parts of the query object to be more readable.
 	 */
 	public function sanitize_ep_query_for_logging( $query ) {
 		if ( isset( $query['args']['headers']['Authorization'] ) ) {
 			$query['args']['headers']['Authorization'] = '<redacted>';
 		}
 
+		// Try to parse the body if possible to make it better readable in the log entry.
+		if ( isset( $query['args']['body'] ) ) {
+			$decoded_body          = json_decode( $query['args']['body'] );
```

Comment:
nothing wrong with this, but my preference is to force the [associative flag](https://www.php.net/manual/en/function.json-decode.php) here, to be absolutely sure that the product is always an associative array and not a generic object.

Reply:
@jesse-greathouse thanks for the comment. Generally speaking, I would agree with you but in this case, it doesn't really matter much as it's used right on the next line and it'll get json_encode'd again when logging and would become an object.